### PR TITLE
Change the schema template to a decorator.

### PR DESCRIPTION
**Pull request by shenshan**

### CODE REVIEW

File: djutils/templates.py [hunk 1]
```diff
@@ -57,9 +55,19 @@ def _check_requirements(self, requirements):
...
-    def init_pipeline(self, schema, requirements=None, context={}, add_here=False):
+    def __call__(self, table_class, context=None):
+        '''
+        While decorating, add table classes into self.tables
+        '''
+        if not context:
+            context = inspect.currentframe().f_back.f_locals
+
+        self._table_classes.append(table_class)
+
```
Comment (eywalker) [inline djutils/templates.py:66]: As a proper decorator, you’d want to return the decorated class, so should do `return table_class`. Also how would the context be passed when used as the decorator?

Comment (dimitri-yatsenko) [reply djutils/templates.py:66]: The SchemaTemplate should follow the same patter of declaration and use as `dj.Schema`. In fact, we discussed to eventually roll this functionality into `dj.Schema`.

Comment (eywalker) [reply djutils/templates.py:66]: Agreed about making it as similar to schema to begin with. In that case, context should be passed into `__init__` in that case.

Comment (shenshan) [reply djutils/templates.py:66]: yes hope to make it similar as `dj.Schema`, but the tables are declared when the users call the method `declare_tables`. This decoration is just to get the list of tables of schema and avoid replication of table listing. I may have missed something conceptionally. Let's discuss tomorrow.

File: djutils/templates.py [hunk 2]
```diff
@@ -83,9 +91,7 @@ def init_pipeline(self, schema, requirements=None, context={}, add_here=False):
             print('Initializing {}'.format(table_class.__name__))
             table = schema(table_class, context=context)
             context[table.__name__] = table
-            tables[table.__name__] = table
+            setattr(self, table.__name__, table)
 
         if add_here:
```
Comment (ttngu207) [inline djutils/templates.py:96]: I think we should remove the `add_here` altogether

File: djutils/templates.py [hunk 3]
```diff
@@ -1,38 +1,36 @@
...
 
     """
-    A pipeline object is a class to initiate a datajoint pipeline based on
-    the template module.
+    A schema object is a decorator for datajoint pipeline classes
     """
 
-    def __init__(self, table_classes,
-                 required_table_names=None,
+    def __init__(self, upstream_table_names=None,
```
Comment (ttngu207) [inline djutils/templates.py:10]: Using this as a decorator, I think we can infer the `upstream_table_names` (checking for Ellipsis object) and `upstream_method_names` based on the decorated class and not needing to be pass in during `__init__`, one less step in usage.

**Pull request merged**